### PR TITLE
QA: Capybara hold the web browser session

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

This code changes the actual behavior of our test suite, currently, we open a new web session for each scenario, so we require a log-in for each scenario. This PR aims to keep open the browser for the whole feature, resetting the session only when a new Cucumber feature starts.

Some scenarios rely on the current behavior, so they expect that we have a reset page on every scenario, but if we change that by holding the session, we must assure that the scenarios don't rely anymore on this.

So, this PR also includes a change in the KVM/XEN tests where we open a new tab, and we expect to have it closed when a new scenario starts. What we do in this case is to assure that we close the tab and we move to the first tab.

**Note:** 
This PR will speed up the test suite "per se", but holding the session by default will help to speed up the tests if we start removing the logout and log-in actions that we do every time we start a scenario. I will proceed to these refactors in next PRs.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- Cucumber tests were changed

- [x] **DONE**

## Links

Ports:
- Manager-4.0
- Manager-4.1

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by srbarrios** [comment]
Trying the switch of tabs, to fix the corner case of not resetting the browser session when you are in a different tab.
As before holding the session, we reset the session, the second tab is closed and you starts from the Uyuni home page.

https://user-images.githubusercontent.com/2827771/114170324-0c14be80-9933-11eb-878c-ca83a58de2e3.mp4

**Comment by srbarrios** [comment]
Testing PR:
https://ci.suse.de/blue/organizations/jenkins/uyuni-pr-ci-tests/detail/uyuni-pr-ci-tests/65/pipeline

File: testsuite/features/secondary/minkvm_guests.feature [hunk 1]
```diff
@@ -82,6 +82,7 @@ Feature: Be able to manage KVM virtual machines via the GUI
     When I click on "Graphical Console" in row "test-vm"
     And I switch to last opened window
     Then I wait until I see the VNC graphical console
```
Comment (Bischoff) [inline testsuite/features/secondary/minkvm_guests.feature:84]: And

File: testsuite/features/secondary/minkvm_guests.feature [hunk 2]
```diff
@@ -223,6 +224,7 @@ Feature: Be able to manage KVM virtual machines via the GUI
     When I click on "Graphical Console" in row "test-vm2"
     And I switch to last opened window
     Then I wait until I see the spice graphical console
```
Comment (Bischoff) [inline testsuite/features/secondary/minkvm_guests.feature:226]: And

File: testsuite/features/secondary/minkvm_guests.feature [hunk 3]
```diff
@@ -430,6 +432,7 @@ Feature: Be able to manage KVM virtual machines via the GUI
     And I wait until I see the VNC graphical console
     Then I wait at most 500 seconds until I see modal containing "Disconnected" text
```
Comment (Bischoff) [inline testsuite/features/secondary/minkvm_guests.feature:433]: And

File: testsuite/features/support/env.rb [hunk 4]
```diff
@@ -103,6 +103,16 @@ def enable_assertions
       debug_server_on_realtime_failure
     end
   end
+  page.instance_variable_set(:@touched, false)
+end
+
+# Reset the page if it's running the first scenario of a feature
+$first_scenario = true
+Before do
```
Comment (Bischoff) [inline testsuite/features/support/env.rb:111]: too bad we don't have hooks at feature level, it would avoid running this for every scenario :-(

**Review by Bischoff — CHANGES_REQUESTED** [review]
minor nitpicks

File: testsuite/features/secondary/minkvm_guests.feature (hunk 1 above)
Comment (srbarrios) [reply testsuite/features/secondary/minkvm_guests.feature:84]: Not part of my changes

File: testsuite/features/secondary/minkvm_guests.feature (hunk 2 above)
Comment (srbarrios) [reply testsuite/features/secondary/minkvm_guests.feature:226]: Not part of my changes

File: testsuite/features/secondary/minkvm_guests.feature (hunk 3 above)
Comment (srbarrios) [reply testsuite/features/secondary/minkvm_guests.feature:433]: Not part of my changes

File: testsuite/features/support/env.rb (hunk 4 above)
Comment (srbarrios) [reply testsuite/features/support/env.rb:111]: yup, let's move to java, there they implemented it :trollface:

**Comment by srbarrios** [comment]
Note:
I will port it only after we had a CI pass and I can assure it will not break other CIs
